src/elements/form/extras/address/Address.js

```diff
@@ -73,17 +82,25 @@ export class NovoAddressElement implements ControlValueAccessor {
 
     updateControl() {
         this.onModelChange(this.model);
+        this.changeDetectorRef.markForCheck();
     }
 
     writeValue(model:any):void {
-        if (!this.model || !this.model.length) {
-            this.model = {
-                countryID: 1,
-                countryName: 'United States'
-            };
-        } else {
-            this.model = model;
-        }
+        setTimeout(() => {
```

Comment:
The data that comes back for address for most entities is:

```
"address": {
"address1": "",
"address2": "",
"city": "Pleasanton",
"state": "CA",
"zip": "94588",
"countryID": 1,
"countryName": "United States                                                   ",
"countryCode": "US"
}
```

While, JobOrder/Opportunity are the only ones that look like

```
"address": {
"address1": "HP-8",
"city": "High Point",
"state": "North Carolina",
"zip": "27261",
"countryID": 1
}
```

In that case, the data is already there and this is doing unnecessary checks. We will still have to handle the state issue (see comment below).

There is a SAT Chat topic on fixing Address fields, since we have three types: Address, Address1, and AddressWithoutCountry.  

I think the solution should be a mixture of what is here and fixing REST.

Plus there is also a ticket in Fit&Finish (or the spreadsheet) to fix required fields and Address. Since sometimes only city/state is required and others everything is.